argmin/src/solver/brent/brentopt.rs

```diff
@@ -97,7 +97,7 @@ impl<F: ArgminFloat> BrentOpt<F> {
     }
 }
 
-impl<O, F> Solver<O, IterState<F, (), (), (), F>> for BrentOpt<F>
+impl<O, F, R> Solver<O, IterState<F, (), (), (), F, R>> for BrentOpt<F>
```

Comment:
Solvers which do not use residuals should have `R = ()`. This applies to most of the changes in the `solver` module.